runtime-interfaces/src/snark_verifier/mod.rs

```diff
@@ -0,0 +1,53 @@
+//! An interface that provides to the runtime a functionality of verifying halo2 SNARKs, together with related errors
+//! and configuration.
+
+#[cfg(feature = "std")]
+mod implementation;
+#[cfg(all(test, feature = "std"))]
+mod tests;
+
+#[cfg(feature = "std")]
+pub use implementation::{Curve, Fr, G1Affine};
+// Reexport `verify` and `HostFunctions`, so that they are not imported like
+// `aleph-runtime-interfaces::snark_verifier::snark_verifier::<>`.
+pub use snark_verifier::verify;
+#[cfg(feature = "std")]
+pub use snark_verifier::HostFunctions;
+
+/// Log_2(max number of rows in a supported circuit).
+///
+/// Note: the same constant MUST be used in the params generation for preparing proving and
+/// verifying keys.
+pub const CIRCUIT_MAX_K: u32 = 12;
```

Comment:
could we maybe add `k` to verifying_key?
And then we don't need this constant here I suppose

Reply:
discussed offline: as it requires small changes to pallet's API, it'll be done in the next PR